e2e/testsuite/tx.go

```diff
@@ -167,15 +167,31 @@ func (s *E2ETestSuite) ExecuteAndPassGovV1Proposal(ctx context.Context, msg sdk.
 
 	s.Require().NoError(cosmosChain.VoteOnProposalAllValidators(ctx, strconv.Itoa(int(proposalID)), cosmos.ProposalVoteYes))
 
+	err = s.waitForGovV1ProposalToPass(ctx, cosmosChain, proposalID)
+
+	s.Require().NoError(err)
+}
+
+// waitForGovV1ProposalToPass polls for the entire voting period to see if the proposal has passed.
+// if the proposal has not passed within the duration of the voting period, an error is returned.
+func (s *E2ETestSuite) waitForGovV1ProposalToPass(ctx context.Context, chain ibc.Chain, proposalID uint64) error {
+	var govProposal govtypesv1.Proposal
 	// poll for the query for the entire voting period to see if the proposal has passed.
-	err = test.WaitForCondition(testvalues.VotingPeriod, 10*time.Second, func() (bool, error) {
-		proposal, err := s.QueryProposalV1(ctx, cosmosChain, proposalID)
+	err := test.WaitForCondition(testvalues.VotingPeriod, 10*time.Second, func() (bool, error) {
+		proposal, err := s.QueryProposalV1(ctx, chain, proposalID)
 		if err != nil {
-			return false, err
+			return false, nil
```

Comment:
wouldn't we still want to return `err` here?

Reply:
Thinking some more we probably do, there should be no reason why the grpc endpoint returns an error in the happy path.